### PR TITLE
Allow arguments supplied on the cli with `tctl` to override profile defaults

### LINKED ISSUE #12124: tctl index out of range panic
### Description

**What happened**:

```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/gravitational/teleport/tool/tctl/common.Run({0xc000b7be80, 0xf, 0x7ff84b0dde70?})
        /tmp/teleport-20220407-3274045-1ke6j8u/teleport-9.0.4/tool/tctl/common/tctl.go:169 +0xf9b
main.main()
        /tmp/teleport-20220407-3274045-1ke6j8u/teleport-9.0.4/tool/tctl/main.go:41 +0x30a
```

**What you expected to happen**:

Not panicking

After applying the following patch, it at least emits a much more reasonable, although still unhelpful, error
> [CLIENT]       Cannot connect to the auth server: failed direct dial to auth server: must provide Dialer or Addrs in config
>        must provide Dialer or Addrs in config.
> Is the auth server running on "UNKNOWN"?

```diff
diff --git a/tool/tctl/common/tctl.go b/tool/tctl/common/tctl.go
index b75b6efa3..554a04bb4 100644
--- a/tool/tctl/common/tctl.go
+++ b/tool/tctl/common/tctl.go
@@ -164,9 +164,13 @@ func Run(commands []CLICommand) {
 
        client, err := authclient.Connect(ctx, clientConfig)
        if err != nil {
+               addr := "UNKNOWN"
+               if len(cfg.AuthServers) > 0 {
+                       addr = cfg.AuthServers[0].Addr
+               }
                utils.Consolef(os.Stderr, log.WithField(trace.Component, teleport.ComponentClient), teleport.ComponentClient,
                        "Cannot connect to the auth server: %v.\nIs the auth server running on %q?",
-                       err, cfg.AuthServers[0].Addr)
+                       err, addr)
                os.Exit(1)
        }
 
```

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. `tctl auth sign --auth-server example.teleport.sh:3025 --out slack_file --format=file --app-name slack-plugin`
2. :bomb:

### Server Details
- Teleport version (run `teleport version`): Teleport v9.0.4 git: go1.18
- Server OS (e.g. from `/etc/os-release`): Ubuntu 21.10
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): N/A
- Additional details:

### Client Details
- Tsh version (`tsh version`):
- Computer OS (e.g. Linux, macOS, Windows):
- Browser version (for UI-related issues):
- Installed via (e.g. apt, yum, brew, website download):
- Additional details:

### Debug Logs
Please include or attach debug logs, when appropriate. Obfuscate sensitive information!
- Start Teleport with --debug flag (`teleport --debug`)
- Run tsh with --debug flag (`tsh --debug`)

### PR DESCRIPTION
**Pull request by hatched**
When supplying additional arguments to `tctl` commands, override the values provided by the profile defaults. 

This could be replicated if providing the `--auth-server` flag when not including a configuration file path. 
ie) `./tctl status --auth-server=my-cluster:3025`

Fixes: #12124

### CODE REVIEW

File: tool/tctl/common/tctl.go [hunk 1]
```diff
@@ -165,9 +165,13 @@ func Run(commands []CLICommand) {
...
 	client, err := authclient.Connect(ctx, clientConfig)
 	if err != nil {
+		addr := "UNKNOWN"
+		if len(cfg.AuthServers) > 0 {
+			addr = cfg.AuthServers[0].Addr
+		}
 		utils.Consolef(os.Stderr, log.WithField(trace.Component, teleport.ComponentClient), teleport.ComponentClient,
 			"Cannot connect to the auth server: %v.\nIs the auth server running on %q?",
-			err, cfg.AuthServers[0].Addr)
+			err, addr)
```
Comment (espadolini) [inline tool/tctl/common/tctl.go:174]: Unless I misunderstood something, this should work (and gets rid of the weird `"UNKNOWN"` thing).

```suggestion
			err, clientConfig.AuthServers[0].Addr)
```

**Review by espadolini — COMMENTED** [review]
The `applyConfig` function is now (maybe even before?) capable of desyncing the `service.Config` and the returned `authclient.Config`; we should decide if the only source of truth should be the returned value (and so the docs for the function should be changed) or if the `cfg` should still be kept in sync.

**Review by r0mant — COMMENTED** [review]
@hatched Could you please add test coverage to this PR?

**Comment by hatched** [comment]
@r0mant yeah you bet, there are some additional changes that need to be made so when I can get back to this I'll be sure to add those tests as well.

**Review by capnspacehook — APPROVED** [review]

**Comment by zmb3** [comment]
Yes, @hatched let's get this across the finish line and merged to master.

Let's not backport it, since it technically changes the behavior and could break existing workflows (unlikely, but you never know).

**Review by espadolini — COMMENTED** [review]
LGTM, waiting on test coverage

**Comment by GavinFrazar** [comment]
Didn't see this when I made #16130. I don't agree that we should skip loading from profile if they provide `--auth-server` flag, because it breaks remote tctl usage unless they also specify `--identity`, right? I think we should just set `cfg.AuthServerAddr` from the cli flag if it is present. @hatched what do you think of my PR's solution to just move the code for loading from cli flag above the profile loading?

**Comment by hatched** [comment]
After a discussion I'm closing this as it's been superseded by https://github.com/gravitational/teleport/pull/16130

**Pull request closed without merging**